Sources/StreamChat/Database/DatabaseContainer.swift

```diff
@@ -152,6 +152,14 @@ class DatabaseContainer: NSPersistentContainer {
             log.debug("Starting a database session.")
             do {
                 try actions(self.writableContext)
+                // If you touch ManagedObject and update one of it properties to same value
+                // Object will be marked as `updated` even it hasn't changed.
+                // By reseting such objects we remove updates that are not updates.
+                for object in self.writableContext.updatedObjects {
+                    if object.changedValues().isEmpty {
+                        self.writableContext.refresh(object, mergeChanges: false)
+                    }
```

Comment:
Do I understand it correctly, that `object.hasChanges` is set whenever you set a value on the object, but `object.changedValues()` actually evaluates the value and returns only when the value was actually changed?

Does it mean that we can actually revert the changes from https://github.com/GetStream/stream-chat-swift/pull/752 ? cc @b-onc

Reply:
If that's ⬆️ the case, we definitely don't need that persisted hash workaround, this would be a much cleaner one.
Also, the existence of this PR proves that #752 did not resolve the issue fully.

Reply:
Basically yes. `hasChanges` is just dirty flag, so whenever you touch object, it gets marked as with changes.
`changedValues` contains pairs of keys-values of actually changed values.
Here exist additional property `hasPersistentChangedValues` that should work as you would expect from `hasChanges`, but documentation is lucking, so I went with current solution